MonoDevelop.MSBuild.Editor/Completion/MSBuildCompletionSource.cs

```diff
@@ -119,19 +125,141 @@ CompletionContext CreateCompletionContext (List<CompletionItem> items)
 
 			foreach (var att in rr.GetAttributeCompletions (doc, doc.ToolsVersion)) {
 				if (!existingAtts.ContainsKey (att.Name)) {
-					items.Add (CreateCompletionItem (att));
+					items.Add (CreateCompletionItem (att, MSBuildSpecialCommitKind.Attribute));
 				}
 			}
 
 			return CreateCompletionContext (items);
 		}
 
-		CompletionItem CreateCompletionItem (BaseInfo info, string prefix = null)
+		//Dictionary<int, string> cache = new Dictionary<int, string> ();
+
+		protected override async Task<CompletionContext> GetAttributeValueCompletionsAsync (
```

Comment:
This entire override is redundant, the expression value completion already handles triggering expressions of type NuGetID (and will handle additional cases such as DotNetToolReference).

Reply:
`GetAttributeValueCompletionsAsync ` function is not been called anymore. In earlier versions of the extension, it was being called. Should I remove it or let it be there?

Reply:
You can remove it, `GetExpressionCompletionsAsync` handles attribute values - a simple value is just a constant MSBuild expression :).

Reply:
Okay! I removed it. You can see my [latest commit](https://github.com/mhutch/MonoDevelop.MSBuildEditor/pull/19/commits/6c6c30e375c322efe8d028eb602abc33130a363b) for all the changes